bookshelf/gce/deploy.sh

```diff
@@ -0,0 +1,151 @@
+# Copyright 2015 Google Inc.
+#d
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#! /bin/bash
+
+set -ex
+
+ZONE=us-central1-f
+
+GROUP=frontend-group
+TEMPLATE=$GROUP-tmpl
+MACHINE_TYPE=f1-micro
+STARTUP_SCRIPT=my-startup.sh
+IMAGE=ubuntu-15-04
+SCOPES="userinfo-email,\
+logging-write,\
+storage-full,\
+datastore,\
+https://www.googleapis.com/auth/projecthosting"
+TAGS=http-server
+
+MIN_INSTANCES=1
+MAX_INSTANCES=10
+TARGET_UTILIZATION=0.6
+
+SERVICE=frontend-web-service
+
+#
+# Instance group setup
+#
+
+# First we have to create an instance template.
+# This template will be used by the instance group
+# to create new instances.
+
+# [START create_template]
+gcloud compute instance-templates create $TEMPLATE \
+  --machine-type $MACHINE_TYPE \
+  --scopes $SCOPES \
+  --metadata-from-file startup-script=$STARTUP_SCRIPT \
+  --image $IMAGE \
+  --tags $TAGS
+# [END create_template]
+
+# Create the managed instance group.
+
+# [START create_group]
+gcloud compute instance-groups managed \
+  create $GROUP \
+  --base-instance-name $GROUP \
+  --size $MIN_INSTANCES \
+  --template $TEMPLATE \
+  --zone $ZONE
+# [END create_group]
+
+# [START create_named_port]
+gcloud compute instance-groups managed set-named-ports \
+    $GROUP \
+    --named-port http:8080 \
+    --zone $ZONE
+# [END create_named_port]
+
+#
+# Load Balancer Setup
+#
+
+# A complete HTTP load balancer is structured as follows:
+#
+# 1) A global forwarding rule directs incoming requests to a target HTTP proxy.
+# 2) The target HTTP proxy checks each request against a URL map to determine the
+#    appropriate backend service for the request.
+# 3) The backend service directs each request to an appropriate backend based on
+#    serving capacity, zone, and instance health of its attached backends. The
+#    health of each backend instance is verified using either a health check.
+#
+# We'll create these resources in reverse order:
+# service, health check, backend service, url map, proxy.
+
+# Create a health check
+# The load balancer will use this check to keep track of which instances to send traffic to.
+# Note that health checks will not cause the load balancer to shutdown any instances.
+
+# [START create_health_check]
+gcloud compute http-health-checks create ah-health-check \
+  --request-path /_ah/health
```

Comment:
I don't know the answer to that, but we should just add a health-check route to silex.